src/go/k8s/pkg/resources/configmap.go

```diff
@@ -138,6 +150,22 @@ func (r *ConfigMapResource) createConfiguration() *config.Config {
 		}
 	}
 
+	if r.pandaCluster.Spec.ArchivalStorage.Enabled {
+		secretName := types.NamespacedName{
+			Name:      r.pandaCluster.Spec.ArchivalStorage.S3SecretKeyRef.Name,
+			Namespace: r.pandaCluster.Spec.ArchivalStorage.S3SecretKeyRef.Namespace,
+		}
+		// We need to retrieve the Secret containing the provided S3 secret key and extract the key itself.
+		s3SecretKeyStr, err := r.getSecretValue(ctx, secretName, r.pandaCluster.Spec.ArchivalStorage.S3SecretKeyRef.Name)
+		if err != nil {
+			return nil, fmt.Errorf("cannot retrieve s3 secret for data archival: %w", err)
+		}
+		if s3SecretKeyStr == "" {
+			return nil, fmt.Errorf("secret name %s, ns %s: %w", secretName.Name, secretName.Namespace, errS3SecretKeyCannotBeEmpty)
```

Comment:
the error should say something like that it's empty perhaps?

Reply:
Thanks for the review.

There's `errS3SecretKeyCannotBeEmpty`, which would print "s3SecretKey string cannot be empty".